### PR TITLE
fix(e2e): refactor e2e tests

### PR DESCRIPTION
**Pull request by moolen**
This PR refactors our e2e test suite and makes the test cases reusable across providers.
note: some use-cases are not supported on some providers, this is documented in the comments.

tl;dr

```go
// SecretStoreProvider is a interface that must be implemented
// by a provider that runs the e2e test
type SecretStoreProvider interface {
	CreateSecret(key string, val string)
	DeleteSecret(key string)
}
```

In an e2e test we must only instantiate a provider and run the test suite like this:

```go
f := framework.New("eso-azure")
prov := newXYZProvider(f, /* credentials, config etc. */)

DescribeTable("sync secrets", framework.TableFunc(f, prov),
  Entry(common.SimpleDataSync(f)),
  Entry(common.NestedJSONWithGJSON(f)),
  Entry(common.JSONDataFromSync(f)),
  Entry(common.JSONDataWithProperty(f)),
  Entry(common.JSONDataWithTemplate(f)),
)
```

### CODE REVIEW

File: e2e/Makefile [hunk 1]
```diff
@@ -5,6 +5,7 @@ SHELL       := /bin/bash
 IMG_TAG     = test
 IMG         = local/external-secrets-e2e:$(IMG_TAG)
 K8S_VERSION = "1.19.1"
+export FOCUS := $(FOCUS)
```
Comment (moolen) [inline e2e/Makefile:8]: This is helpful to focus a certain testsuite, e.g.:

```sh
make test.e2e FOCUS="(aws|vault|azure)"
```

**Comment by knelasevero** [comment]
Thats beautiful :sparkles:

**Comment by knelasevero** [comment]
I was hoping for a gif a bit more celebrational :joy:

**Pull request merged**